optuna/samplers/_tpe/sampler.py

```diff
@@ -516,7 +550,29 @@ def after_trial(
         state: TrialState,
         values: Optional[Sequence[float]],
     ) -> None:
-
+        assert state in [TrialState.COMPLETE, TrialState.FAIL, TrialState.PRUNED]
```

Comment:
[Question] Could you tell me why this assertion is added? The following lines also checks the state in if-statement, and I guess it might not be necessary.

Reply:
Ah, it follows the implementation of NSGAIISampler.
https://github.com/optuna/optuna/blob/83c8022bcb3847b261e3f0b65aeaccf46dbb8cb6/optuna/samplers/nsgaii/_sampler.py#L391

Reply:
Yeah, I just followed NSGAIISampler without much thought. Do you think I should remove this line?

Reply:
I just wanted to know the reason. I think we can keep it. Thank you!